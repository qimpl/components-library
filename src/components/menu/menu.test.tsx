import React from 'react';

import { render } from '@testing-library/react';

import Menu from '.';
import MenuItem from './menuItem';

test('Default menu', () => {
  expect(
    render(
      <Menu toggleButtonContent='open menu'>
        <>
          <MenuItem>Item 1</MenuItem>
          <MenuItem>
            <p>Item 2</p>
          </MenuItem>
        </>
      </Menu>
    )
  ).toMatchSnapshot();
});

test('Dropdown menu', () => {
  expect(
    render(
      <Menu toggleButtonContent='open menu' isDropdown>
        <>
          <MenuItem>
            <p>Item 1</p>
          </MenuItem>
          <MenuItem>
            <p>Item 2</p>
          </MenuItem>
        </>
      </Menu>
    )
  ).toMatchSnapshot();
});
