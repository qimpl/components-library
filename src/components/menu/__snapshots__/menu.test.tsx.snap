// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default menu 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gtsrHT fXFjuR"
      >
        <button
          class="sc-bdnxRM dEuXiE"
          color="#013c63"
          type="button"
        >
          open menu
        </button>
        <ul
          class="sc-dlnjwi epeCoM"
        >
          <li
            class="sc-hKFxyN bhsPSn"
          >
            Item 1
          </li>
          <li
            class="sc-hKFxyN bhsPSn"
          >
            <p>
              Item 2
            </p>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gtsrHT fXFjuR"
    >
      <button
        class="sc-bdnxRM dEuXiE"
        color="#013c63"
        type="button"
      >
        open menu
      </button>
      <ul
        class="sc-dlnjwi epeCoM"
      >
        <li
          class="sc-hKFxyN bhsPSn"
        >
          Item 1
        </li>
        <li
          class="sc-hKFxyN bhsPSn"
        >
          <p>
            Item 2
          </p>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Dropdown menu 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gtsrHT fXFjuR"
      >
        <button
          class="sc-bdnxRM dEuXiE"
          color="#013c63"
          type="button"
        >
          open menu
        </button>
        <ul
          class="sc-dlnjwi jjCIwR"
        >
          <li
            class="sc-hKFxyN bhsPSn"
          >
            <p>
              Item 1
            </p>
          </li>
          <li
            class="sc-hKFxyN bhsPSn"
          >
            <p>
              Item 2
            </p>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gtsrHT fXFjuR"
    >
      <button
        class="sc-bdnxRM dEuXiE"
        color="#013c63"
        type="button"
      >
        open menu
      </button>
      <ul
        class="sc-dlnjwi jjCIwR"
      >
        <li
          class="sc-hKFxyN bhsPSn"
        >
          <p>
            Item 1
          </p>
        </li>
        <li
          class="sc-hKFxyN bhsPSn"
        >
          <p>
            Item 2
          </p>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
